### ISSUE TITLE
Removing MyBB Default Capatcha 

### ISSUE THREAD

**Issue by Ben-MyBB** [mybb/mybb#3522]
It was recently decided that many users would support us removing the MyBB Default Capatcha with a view to keeping up-to-date with Google's releases.

We did also think it would be a good idea to include steps in the Installation guide to ask for a Google API key.

**Comment by labrocca** [comment]
I like default MYBB Captcha. No key required and no updating needed.  It's a working feature, why remove it?

**Comment by kawaii** [comment]
The default MyBB CAPTCHA is _beyond_ broken, to call it a "working feature" is beyond laughable too. It offers board owners nothing but a false sense of security.

I'm open to us **improving** (read: replacing) the default CAPTCHA with an equally convenient alternative that doesn't require a key and runs out of the box - but that's easy to say.

Google reCAPTCHA is a tried and tested replacement which takes just a few minutes to configure and deploy until the time if and when someone develops a homebrew solution better than what we currently have.

**Comment by labrocca** [comment]
When you say it's "broken", please explain because I use it on all my sites and haven't experienced any real problems.  And my guess is that using reCaptcha isn't any easier.  Their captcha can be extremely annoying which is why I try not to use it.  

Using a 3rd party JS, even Google, imho can represent problems.  I have less control and they have full control. Example is when they deprecated early versions forcing everyone to upgrade. I still visit sites with the old version and now you can't use the form as its broken. 

I think adding it as an option was a great idea. But making it the only choice I don't agree with. 

You mentioned replacing it with something homebrew.  

btw, want to discuss this with you more in XMPP.

**Comment by dominikdalek** [comment]
Moving over to the Google reCAPTCHA means that there has to be a working toggle that disables it.
1. Forum won't work offline (i.e on intranet) with reCAPTCH blocking actions.
2. Forum will have to use certificate (reCAPTCHA  doesn't work over HTTP any more).
3. People behind VPN, Tor users, and people using private browsing features of modern browsers will have harder time accessing forum or won't be able to access it at all.

There are commercial BBs that got this very wrong and disabling reCAPTCHA actually meant hiding it (but the script was still active), effectively making forum unusable.

I understand that bots are a problem but I'd rather not use any third party captcha and either have option to plug mine in or to use challenge-response field as a means of verification.

**Comment by euantorano** [comment]
Challenge response is already built in. reCAPTCHA won’t be mandatory - you can turn it on or off fully at your discretion.

**Comment by dvz** [comment]
IMO we should have at least one local implementation (possibly a dependency to replace own solution).

**Comment by euantorano** [comment]
I have no problem including a local built in captcha. If nothing else, it means we can have a captcha enabled out of the box with no extra configuration required. I would like to see the captcha support audio or something though for users who are hard of sight.

**Comment by Eldenroot** [comment]
Yes, we should have one offline solution... maybe we can use some different 3rd party library?

**Comment by arcalys** [comment]
Would you guys consider using https://github.com/Gregwar/Captcha?

**Comment by Sama34** [comment]
I agree a local implementation would be the preferred default. A dependency won't be bad either to save development resources.

@arcalys seems nice, what about the Symfony bundle ? It seems to rely on other Symfony components that may not be used by the core at the moment but seem like reasonable components to implement later anyway, so we could as well go with the bundle from the beginning.

**Comment by arcalys** [comment]
How do you want to handle this for now? Should we replace the contents of `class_captcha.php`, `xmlhttp.php`, and `captcha.php` with the new Captcha and keep the current structure for now (possibly refactoring in 1.10), or should we go ahead and refactor the Captcha into `inc/src` right away, starting with the new default Captcha?

Got a POC running with Gregwar/Captcha:

<img width="2545" height="373" alt="Image" src="https://github.com/user-attachments/assets/bc3344e9-ad5f-4a10-b03c-05e4ad96e9b7" />
